File: disk-ann/src/main/java/jetbrains/exodus/diskann/RobustPruneVertex.java
package jetbrains.exodus.diskann;

final class RobustPruneVertex implements Comparable<RobustPruneVertex> {
    final int index;
    final float distance;

    RobustPruneVertex(int index, float distance) {
        this.index = index;
        this.distance = distance;
    }

    public int compareTo(RobustPruneVertex other) {
        return Double.compare(distance, other.distance);
    }
}

File: disk-ann/src/main/kotlin/jetbrains/exodus/diskann/RobustPruneVertex.kt
package jetbrains.exodus.diskann

internal class RobustPruneVertex(@JvmField val index: Long, @JvmField val vector: FloatArray,
                                 @JvmField val distance: Double) : Comparable<RobustPruneVertex> {
    override fun compareTo(other: RobustPruneVertex): Int {
        return distance.compareTo(other.distance)
    }
}